src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVLinkage.java

```diff
@@ -182,35 +192,77 @@ private static boolean clusterTogetherWithParams(final SVCallRecord a, final SVC
         }
     }
 
+    /**
+     * Performs overlap testing on each pair of complex intervals in two records, requiring each pair to be
+     * sufficiently similar by reciprocal overlap, size similarity, and breakend proximity.
+     */
+    private static boolean testComplexIntervals(final SVCallRecord a, final SVCallRecord b, final double overlapThreshold,
+                                                final double sizeSimilarityThreshold, final int window,
+                                                final SAMSequenceDictionary dictionary) {
+        final List<SVCallRecord.ComplexEventInterval> intervalsA = a.getComplexEventIntervals();
+        final List<SVCallRecord.ComplexEventInterval> intervalsB = b.getComplexEventIntervals();
+        if (intervalsA.size() != intervalsB.size()) {
+            return false;
+        }
+        final Iterator<SVCallRecord.ComplexEventInterval> iterA = intervalsA.iterator();
+        final Iterator<SVCallRecord.ComplexEventInterval> iterB = intervalsB.iterator();
+        for (int i = 0; i < intervalsA.size(); i++) {
+            final SVCallRecord.ComplexEventInterval cpxIntervalA = iterA.next();
+            final SVCallRecord.ComplexEventInterval cpxIintervalB = iterB.next();
+            if (cpxIntervalA.getIntervalType() != cpxIintervalB.getIntervalType()) {
```

Comment:
I'm wondering if you should also check the CPX_TYPE? Although I think this is sufficient, because even though you could technically get a dupINV and a dDUP with the same CPX_INTERVALS, they would have different CHROM and POS

Reply:
Right, `typesMatch()` looks at the subtypes.